File: python/gaia/dependencymapsubview.py
# -*- coding: utf-8 -*-

from pyelemental import RESTObject
from pyelemental import validate_string_in_list, validate_float_in_list, validate_int_in_list, validate_required_int, validate_required_float, validate_required_string, validate_required_time, validate_maximum_float, validate_minimum_float, validate_maximum_int, validate_minimum_int, validate_maximum_length, validate_minimum_length, validate_pattern


class DependencyMapSubview(RESTObject):
    """ Represents a DependencyMapSubview in the 

        Notes:
            A DependencyMapSubview is subview of a DependencyMapView
    """

    def __init__(self, **kwargs):
        """ Initializes a DependencyMapSubview instance

          Notes:
              You can specify all parameters while calling this methods.
              A special argument named `data` will enable you to load the
              object from a Python dictionary

          Examples:
              >>> dependencymapsubview = DependencyMapSubview(id=u'xxxx-xxx-xxx-xxx', name=u'DependencyMapSubview')
              >>> dependencymapsubview = DependencyMapSubview(data=my_dict)
        """

        super(DependencyMapSubview, self).__init__()

        # Read/Write Attributes
        
        self._id = None
        self._selector = None
        self._subselectors = None
        self._tonality = None
        
        self.expose_attribute(local_name="ID", remote_name="ID")
        self.expose_attribute(local_name="selector", remote_name="selector")
        self.expose_attribute(local_name="subSelectors", remote_name="subSelectors")
        self.expose_attribute(local_name="tonality", remote_name="tonality")

        self._compute_args(**kwargs)

    def __str__(self):
        return '<%s:%s>' % (self.identity()["name"], self.identifier)

    def identifier(self):
        """ Identifier returns the value of the object's unique identifier.
        """
        
        return self.ID
        

    def setIdentifier(self, ID):
        """ SetIdentifier sets the value of the object's unique identifier.
        """
        
        self.ID = ID
        

    def identity(self):
        """ Identity returns the Identity of the object.
        """
        return dependencymapsubviewIdentity

    # Properties
    @property
    def ID(self):
        """ Get ID value.

          Notes:
              ID is the identifier of the object.

              
        """
        return self._id

    @ID.setter
    def ID(self, value):
        """ Set ID value.

          Notes:
              ID is the identifier of the object.

              
        """
        self._id = value
    
    @property
    def selector(self):
        """ Get selector value.

          Notes:
              Selector is the main selector for the DependencyMapSubview.

              
        """
        return self._selector

    @selector.setter
    def selector(self, value):
        """ Set selector value.

          Notes:
              Selector is the main selector for the DependencyMapSubview.

              
        """
        self._selector = value
    
    @property
    def subSelectors(self):
        """ Get subSelectors value.

          Notes:
              SubSelectors are the selector to apply inside the main selector.

              
        """
        return self._subselectors

    @subSelectors.setter
    def subSelectors(self, value):
        """ Set subSelectors value.

          Notes:
              SubSelectors are the selector to apply inside the main selector.

              
        """
        self._subselectors = value
    
    @property
    def tonality(self):
        """ Get tonality value.

          Notes:
              Tonality sets the color tonality to use for the DependencyMapSubView.

              
        """
        return self._tonality

    @tonality.setter
    def tonality(self, value):
        """ Set tonality value.

          Notes:
              Tonality sets the color tonality to use for the DependencyMapSubView.

              
        """
        self._tonality = value
    
    def validate(self):
        """ Validate valides the current information stored into the structure.
        """
        errors = []

        if len(errors) > 0:
            return errors

        return None

    # dependencymapsubviewIdentity represents the Identity of the object
dependencymapsubviewIdentity = {"name": "dependencymapsubview", "category": "dependencymapsubviews", "constructor": DependencyMapSubview}

File: golang/dependencymapsubview.go
package gaia

import "fmt"
import "github.com/aporeto-inc/elemental"

// DependencyMapSubviewIdentity represents the Identity of the object
var DependencyMapSubviewIdentity = elemental.Identity{
	Name:     "dependencymapsubview",
	Category: "dependencymapsubviews",
}

// DependencyMapSubviewsList represents a list of DependencyMapSubviews
type DependencyMapSubviewsList []*DependencyMapSubview

// DependencyMapSubview represents the model of a dependencymapsubview
type DependencyMapSubview struct {
	// Selector is the main selector for the DependencyMapSubview.
	Selector []string `json:"selector" cql:"selector,omitempty" bson:"selector"`

	// SubSelectors are the selector to apply inside the main selector.
	SubSelectors map[string][]string `json:"subSelectors" cql:"subselectors,omitempty" bson:"subselectors"`

	// Tonality sets the color tonality to use for the DependencyMapSubView.
	Tonality string `json:"tonality" cql:"tonality,omitempty" bson:"tonality"`
}

// NewDependencyMapSubview returns a new *DependencyMapSubview
func NewDependencyMapSubview() *DependencyMapSubview {

	return &DependencyMapSubview{
		Selector:     []string{},
		SubSelectors: map[string][]string{},
	}
}

// Identity returns the Identity of the object.
func (o *DependencyMapSubview) Identity() elemental.Identity {

	return DependencyMapSubviewIdentity
}

// Identifier returns the value of the object's unique identifier.
func (o *DependencyMapSubview) Identifier() string {

	return ""
}

// SetIdentifier sets the value of the object's unique identifier.
func (o *DependencyMapSubview) SetIdentifier(ID string) {

}

func (o *DependencyMapSubview) String() string {

	return fmt.Sprintf("<%s:%s>", o.Identity().Name, o.Identifier())
}

// Validate valides the current information stored into the structure.
func (o *DependencyMapSubview) Validate() error {

	errors := elemental.Errors{}

	if len(errors) > 0 {
		return errors
	}

	return nil
}

// SpecificationForAttribute returns the AttributeSpecification for the given attribute name key.
func (DependencyMapSubview) SpecificationForAttribute(name string) elemental.AttributeSpecification {

	return DependencyMapSubviewAttributesMap[name]
}

// AttributeSpecifications returns the full attribute specifications map.
func (DependencyMapSubview) AttributeSpecifications() map[string]elemental.AttributeSpecification {

	return DependencyMapSubviewAttributesMap
}

// DependencyMapSubviewAttributesMap represents the map of attribute for DependencyMapSubview.
var DependencyMapSubviewAttributesMap = map[string]elemental.AttributeSpecification{
	"Selector": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Description:    `Selector is the main selector for the DependencyMapSubview.`,
		Exposed:        true,
		Name:           "selector",
		Orderable:      true,
		Stored:         true,
		SubType:        "dependencymapview_selector",
		Type:           "external",
	},
	"SubSelectors": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Description:    `SubSelectors are the selector to apply inside the main selector.`,
		Exposed:        true,
		Name:           "subSelectors",
		Orderable:      true,
		Stored:         true,
		SubType:        "dependencymapview_subselector",
		Type:           "external",
	},
	"Tonality": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Description:    `Tonality sets the color tonality to use for the DependencyMapSubView.`,
		Exposed:        true,
		Filterable:     true,
		Format:         "free",
		Name:           "tonality",
		Orderable:      true,
		Stored:         true,
		Type:           "string",
	},
}
